### DISCUSSION TITLE
What are normal loading times for nextcloud?

### DISCUSSION THREAD

**Discussion by DenuxPlays** [nextcloud/all-in-one#2654] [Questions] [answered]
Hey,
i feel like the loading times  of my nextcloud should be way faster.
For example I need around 10s to get my dashboard to load. (see picture)
![image](https://github.com/nextcloud/all-in-one/assets/83671398/2c72c3d3-92b4-4793-bc54-c89a6a78e725)
And the general feeling when browsing trough folders etc. is not that great.
(Just loading in the files tab takes like 6 to 7 seconds)
I just want to ask if this is normal or not?

The Server should be more than enough:
Intel Xeon D1521
32GB DDR4 ECC RAM
And 2x 500GB SSDs

If this is not normal behavior then I suspect that something on my server is not working as I do not think that this has anything todo with all-in-one.

**Comment by szaimen** [comment]
Hi, the hardware looks good. Additional debugging can take place on the login page in a new incognito window. How long does it take to load for you? With AIO v6.0.0 we made some improvements to js compression. With that it should load almost instantly.

**Reply by DenuxPlays** [reply]
> Hi, the hardware looks good. Additional debugging can take place on the login page in a new incognito window. How long does it take to load for you? With AIO v6.0.0 we made some improvements to js compression. With that it should load almost instantly.

Yeah I already updated to v6.0.0 because of the compression but the log in page needs around 3-5 seconds to load.

**Reply by szaimen** [reply]
> Yeah I already updated to v6.0.0 because of the compression but the log in page needs around 3-5 seconds to load.

How fast is your internet? (upload on server and download on client) That could also be a limiting factor here which might explain 3-5s load time on the login page...

**Reply by DenuxPlays** [reply]
> > Yeah I already updated to v6.0.0 because of the compression but the log in page needs around 3-5 seconds to load.
> 
> 
> 
> How fast is your internet? (upload on server and download on client) That could also be a limiting factor here which might explain 3-5s load time on the login page...

Server: 200Mbits/s (up and down)
Client: 100Mbits/s down and 60Mbits/s up

**Reply by szaimen** [reply]
I see so connection speed should also not be a problem.

**Reply by DenuxPlays** [reply]
> I see so connection speed should also not be a problem.

Yes...
When I am back home I try a fresh installation on my server maybe this helps.

**Reply by szaimen** [reply]
> When I am back home I try a fresh installation on my server maybe this helps.

I am not completely sure about that. Can you follow the guide below?

**Comment by szaimen** [comment]
Did you already follow https://github.com/nextcloud/all-in-one/discussions/2475?

**Reply by DenuxPlays** [reply]
Not yet

**Reply by szaimen** [reply]
Then I would highly recommend that.

**Reply by DenuxPlays** [reply]
> Then I would highly recommend that.

Followed it and it hasn't changed anything.
Also the public ip was always listed...

**Reply by szaimen** [reply, marked as answer]
Okay then the loading times are peobably normal.

**Reply by DenuxPlays** [reply]
> Okay then the loading times are peobably normal.

Okay
Thank you :)

**Comment by szaimen** [comment]
Also are you running AIO behind a reverse proxy?

**Reply by DenuxPlays** [reply]
Yes.
I use caddy as my webserver

**Reply by szaimen** [reply]
Okay, caddy should not have pitfails in terms of performance. So the RP can be excluded in terms of factors.

**Comment by szaimen** [comment]
And did you already see https://github.com/nextcloud/all-in-one/discussions/1335?

**Reply by DenuxPlays** [reply]
Yes that was the first thing I saw/tried but that hasn't really changed much.

**Reply by szaimen** [reply]
> Deactivate all optional addons and apps that you do not really need

Might be an addditional step for debuggin purposes.

**Reply by DenuxPlays** [reply]
> > Deactivate all optional addons and apps that you do not really need
> 
> Might be an addditional step for debuggin purposes.

Already did that as I followed the guide you provided before asking here.

**Comment by k4my4b** [comment]
> Hey, i feel like the loading times of my nextcloud should be way faster. For example I need around 10s to get my dashboard to load. (see picture) ![image](https://user-images.githubusercontent.com/83671398/242609225-2c72c3d3-92b4-4793-bc54-c89a6a78e725.png) And the general feeling when browsing trough folders etc. is not that great. (Just loading in the files tab takes like 6 to 7 seconds) I just want to ask if this is normal or not?
> 
> The Server should be more than enough: Intel Xeon D1521 32GB DDR4 ECC RAM And 2x 500GB SSDs
> 
> If this is not normal behavior then I suspect that something on my server is not working as I do not think that this has anything todo with all-in-one.

In my experience most (95%) of nc problems are related to its frontend, to test things for yourself you can either hook into the api and test things (measure response times) that way or a much simpler way would be to use a different interface like the android/ios app to see if everything loads faster there (eyeballing). 

---- WARNING: Rant incoming ----

The backend api responds pretty fast (fast enough) usually but the front-end is a right mess, each part is written using different tools and frameworks anything from deprecated jquery versions to vuejs and anything in between. Obviously it's the teams choice to go with whatever framework that works for them but vuejs over reactjs or some other jsx based framework (Qwik FTW :heart:) significantly limits the number of people that would be able and willing to contribute to the project, I for one I cannot find the time nor can I bring myself to (despite my best efforts) to learn vuejs just to write nextcloud apps so i guess it is what it is.

**Reply by szaimen** [reply]
This is the wrong place to discuss this. Feel free to open a new thread on https://help.nextcloud.com

**Comment by k4my4b** [comment]
to measure actual response times (files api (webdav)):
```
curl -o /dev/null -s -w "%{time_total}s\n" "https://<yourcloud>/remote.php/dav/files/<user e.g. admin>" --user "<user>:<password*>" --request PROPFIND
```
\* you'll  most likely need to generate an app password for this under `Personal settings > Security > Devices & sessions`

**Reply by DenuxPlays** [reply]
![image](https://github.com/nextcloud/all-in-one/assets/83671398/04aa792e-bae6-4024-a729-fe227a5d6197)
That was the result...
Seems pretty fast to me

**Reply by k4my4b** [reply]
yup 0.8 seconds is alright specially if you're doing this remotely, so now you know where the bottleneck is.

**Reply by DenuxPlays** [reply]
but I dont know how to resolve it
or is the frontend really that slow?

**Reply by k4my4b** [reply]
> but I dont know how to resolve it

Labours of Hercules

> or is the frontend really that slow?

Yarp.
Pop the bonnet (press f12) and see what's going on. 

Post some stuff (logs, screenshots etc... but trim it) here perhaps we can be of some assistance then.
